app/metrics.yaml

```diff
@@ -7301,6 +7301,96 @@ search_terms:
       - android-probes@mozilla.com
     expires: 108
 
+messaging:
+  message_shown:
+    type: event
+    description: |
+      A message was shown to the user.
+    extra_keys:
+      message_key:
+        description: The id of the message
+        type: string
+    bugs:
+      - https://github.com/mozilla-mobile/fenix/issues/24224
+    data_reviews:
+      - https://github.com/mozilla-mobile/fenix/issues/24224
+    notification_emails:
+      - android-probes@mozilla.com
+    data_sensitivity:
+      - interaction
+    expires: 114
+  message_dismissed:
+    type: event
+    description: |
+      A message was dismissed by the user.
+    extra_keys:
+      message_key:
+        description: The id of the message
+        type: string
+    bugs:
+      - https://github.com/mozilla-mobile/fenix/issues/24224
+    data_reviews:
+      - https://github.com/mozilla-mobile/fenix/issues/24224
+    notification_emails:
+      - android-probes@mozilla.com
+    data_sensitivity:
+      - interaction
+    expires: 114
+  message_clicked:
+    type: event
+    description: |
+      A message was clicked by the user.
+    extra_keys:
+      message_key:
+        description: The id of the message
+        type: string
+      action_uuid:
+        description: The uuid of the action
+        type: string
+    bugs:
+      - https://github.com/mozilla-mobile/fenix/issues/24224
+    data_reviews:
+      - https://github.com/mozilla-mobile/fenix/issues/24224
+    notification_emails:
+      - android-probes@mozilla.com
+    data_sensitivity:
+      - interaction
+    expires: 114
+  message_expired:
+    type: event
+    description: |
+      A message maxDisplayCount has been surpassed.
+    extra_keys:
+      message_key:
+        description: The id of the message
+        type: string
+    bugs:
+      - https://github.com/mozilla-mobile/fenix/issues/24224
+    data_reviews:
+      - https://github.com/mozilla-mobile/fenix/issues/24224
+    notification_emails:
+      - android-probes@mozilla.com
+    data_sensitivity:
+      - interaction
+    expires: 114
+  malformed:
```

Comment:
why not ```message-malformed```?

Reply:
I got an error from the glean tooling indicating, that all pings from a category shouldn't have the same prefix. If it's easier, I could should just remove the `message-` prefix for all other events? :)

>   ❌ Found 1 errors.
>   ERROR running glean_parser v5.0.1
>   Sorry, Glean found some glinter nits:
>   ERROR: COMMON_PREFIX: messaging: Within category 'messaging', all metrics begin with prefix 'message'.Remove the prefixes on the metric names and (possibly) rename the category.
>

Reply:
Yes, the "category" of messaging is causing the glean_parser linter (the glinter) to complain that the names of the specific metrics in that category are redundant.

When you go to query something like this in analysis you would be selecting like `SELECT messaging_message_malformed`.  One suggestion would be to invert the naming, so instead of "message_malformed", maybe "malformed_message" is okay? Either that, or you can disable this specific lint if the naming should stay as it is.  If we have already collected data with existing naming, then I recommend that we do not change the names unless we really need to.